src/nupic/algorithms/ColumnPooler.cpp

```diff
@@ -184,10 +191,10 @@ class ColumnPooler // : public Serializable
 
     // Setup the distal dendrites
     distalConnections.initialize(
-        /* columnDimensions */            {2048},
+        /* columnDimensions */            cellDimensions,
```

Comment:
:+1: 
Are you considering the constructor, where `*tm` is passed in? (that would be instead of the Parameters noDistalConncetionsNeeded)